pandas/core/frame.py

```diff
@@ -3392,7 +3392,13 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
             if not labels.is_lexsorted():
                 labels = MultiIndex.from_tuples(labels.values)
 
-            indexer = lexsort_indexer(labels.labels, orders=ascending,
```

Comment:
as I said, you can simply create a method on ``MultiIndex``, called ``_lexsort_indexer`` (let's make it private), that returns the correct indexer.